File: Laptop.java
public class Laptop {
    protected String name;
    protected int ScreenSize;

    public Laptop(String name, int ScreenSize) {
        this.name = name;
        this.ScreenSize = ScreenSize;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getScreenSize() {
        return ScreenSize;
    }

    public void setScreenSize(int ScreenSize) {
        this.ScreenSize = ScreenSize;
    }

    @Override
    public String toString() {
        return "Laptop: " + name + "\n" + "Screen Size: " + ScreenSize;
    }

    public static void main(String[] args) {
        Laptop macbookAir = new MacbookAir(28, 2017);
        System.out.println(macbookAir);
    }
}

class MacbookAir extends Laptop {
    private boolean hasTouchID;

    public MacbookAir(int ScreenSize, int YearOfModel) {
        super("Macbook Air", ScreenSize);
        this.hasTouchID = YearOfModel >= 2018;
    }

    public boolean isHasTouchID() {
        return hasTouchID;
    }

    public void setHasTouchID(boolean hasTouchID) {
        this.hasTouchID = hasTouchID;
    }

    @Override
    public String toString() {
        return name + "\n" + "Screen Size: " + ScreenSize + "\n" + "Has Touch ID: " + hasTouchID;
    }

}

File: Laptop.py
class Laptop:
    def __init__(self, name, screen_size):
        self.name = name
        self.screen_size = screen_size
    
    def __str__(self):
        return f"Laptop name: {self.name} with screen size {self.screen_size} inches"

class MacbookAir(Laptop): 
        def __init__(self, name, screen_size, year_model):
             super().__init__(name, screen_size)
             self.has_touch_ID = year_model >= 2018
        
        def __str__(self):
             return f"{super().__str__()} and TouchID: {self.has_touch_ID}"
        
macbook_air = MacbookAir("Macbook Air", 28, 2017)
print(macbook_air)
        